thruster_control/src/thruster_control.cpp

```diff
@@ -105,65 +107,85 @@ ThrusterControl::ThrusterControl(ros::NodeHandle& nodeHandle)
+  {
+      diagnostic_tools::Diagnostic::WARN, ReportFault::THRUSTER_TEMP_STAGNATED
+  };
+  paramsTemperatureMessageStagnationcheck.stagnation_diagnostic(diagnosticTemperatureInfoStagnate);
 
   diagnosticsUpdater.add(
       publisher_reportMotorTemp.add_check<diagnostic_tools::MessageStagnationCheck>(
           "stagnation check",
           [motorTemperatureSteadyBand](const thruster_control::ReportMotorTemperature& a,
-                                       const thruster_control::ReportMotorTemperature& b) {
+                                       const thruster_control::ReportMotorTemperature& b)
+          {
             return std::fabs(a.motor_temp - b.motor_temp < motorTemperatureSteadyBand);
-          },
-          diagnostic_tools::MessageStagnationCheckParams{}.stagnation_diagnostic(
-              {diagnostic_tools::Diagnostic::WARN,
-               health_monitor::ReportFault::THRUSTER_TEMP_STAGNATED})));
+          }
+          , paramsTemperatureMessageStagnationcheck));
 
   canIntf.SetMotorTimeoutSeconds(setRPMTimeout);
   canIntf.SetEnableCANLogging(currentLoggingEnabled);
 
   motorRPMCheck = diagnostic_tools::create_health_check<double>(
-      "Motor RPM check", [this](double rpms) -> diagnostic_tools::Diagnostic {
+      "Motor RPM check", [this](double rpms) -> diagnostic_tools::Diagnostic
+      {
         using diagnostic_tools::Diagnostic;
-        if (std::abs(rpms) > maxAllowedMotorRPM) {
-          return Diagnostic{Diagnostic::WARN,
-                            health_monitor::ReportFault::THRUSTER_RPM_THRESHOLD_REACHED}
+        if (std::abs(rpms) > maxAllowedMotorRPM)
+        {
+          return Diagnostic
+          {
+            Diagnostic::WARN, ReportFault::THRUSTER_RPM_THRESHOLD_REACHED
+          }
               .description("Absolute RPMs above threshold: |%f| > %f", rpms, maxAllowedMotorRPM);
         }
         return Diagnostic::OK;
-      });
+      }
+);
   diagnosticsUpdater.add(motorRPMCheck);
 
   motorTemperatureCheck = diagnostic_tools::create_health_check<double>(
-      "Motor temperature check", [this](double temperature) -> diagnostic_tools::Diagnostic {
+      "Motor temperature check", [this](double temperature) -> diagnostic_tools::Diagnostic
+      {
         using diagnostic_tools::Diagnostic;
         if (temperature > motorTemperatureThreshold)
         {
-          return Diagnostic{Diagnostic::ERROR,
-                            health_monitor::ReportFault::THRUSTER_TEMP_THRESHOLD_REACHED}
-              .description(
-                  "Temperature above threshold: %f degC > %f degC",
-                  temperature, motorTemperatureThreshold);
+          return Diagnostic
+          {
+            Diagnostic::ERROR, ReportFault::THRUSTER_TEMP_THRESHOLD_REACHED
+          }
+              .description("Temperature above threshold: %f degC > %f degC", temperature,
+                           motorTemperatureThreshold);
```

Comment:
@patapufate meta nit: I'm starting to think `roslint` was configured for pre-C++11 code. Something like: 
```suggestion
          return Diagnostic(Diagnostic::ERROR, ReportFault::THRUSTER_TEMP_THRESHOLD_REACHED)
              .description("Temperature above threshold: %f degC > %f degC", temperature,
                           motorTemperatureThreshold);
```
would read better and work equally well. Same elsewhere. These "open" uniform initialization statements are awful.

Reply:
Ok